tests/junit.py

```diff
@@ -109,6 +119,10 @@ def check_pre_skip(self, test: str, spec: TestSpec, resource: str, nproc: int) -
         for condition in spec.only:
             if (condition == 'cpu') and ('gpu' in resource):
                 return 'CPU only test with GPU backend'
+            if (condition == 'int32') and ('int64' in self.petsc_arch):
+                return 'int32 test with int64 PETSc arch'
+            if (condition == 'int64') and ('int64' not in self.petsc_arch):
+                return 'int64 test with int32 PETSc arch'
```

Comment:
This is quite fragile. It's enough for CI, but not for general developers. Note that `PETSC_ARCH` does not need to be set (and the name has no special meaning). I think a better way would be to check
```console
$ grep USE_64BIT_INDICES $PETSC_DIR/$PETSC_ARCH/include/petscconf.h
#define PETSC_USE_64BIT_INDICES 1
```
This line won't exist for a 32-bit build.

Reply:
I can update this in Ratel too